### ISSUE TITLE
Metric values not matching for pipe test set

### ISSUE THREAD

**Issue by rathoreyuvraj** [RotsteinNoam/Paint-by-Inpaint#4]
Hi, I wanted to compare my method results with yours and I am using CLIP-I, DINO, L1 and L2 metrics as mentioned in your paper. The values for your method are matching for L1 and L2 but not for CLIP-I and DINO. I am using clip-VIT-B-32 and dino-vitb16 between the target images mentioned in your test set and the output generated following your github Readme. Can you clarify am I doing it wrong or I should use different clip and dino models.

**Comment by RotsteinNoam** [comment]
Thanks for checking! Could you share which differences you see?

**Comment by rathoreyuvraj** [comment]
| | CLIP-I| DINO| L1| L2|
|--------|--------|--------|--------|--------|
| Calculated| .928| .884| .057| .014|
| Values mentioned in paper| .945| .903| .057| .014|

**Comment by navvewas** [comment]
Hi!
I think the differences are most likely due to variations in image preprocessing. Could you share the preprocessing steps you used before the CLIP-I and DINO evaluation?

For reference, here’s what we used:
**CLIP-I:** the default transformation from clip.load("ViT-B/32", device)
**DINO** (following the MagicBrush code):
transforms.Resize(256, interpolation=3),
transforms.CenterCrop(224),
transforms.ToTensor(),
transforms.Normalize((0.485, 0.456, 0.406), (0.229, 0.224, 0.225))

**Comment by rathoreyuvraj** [comment]
|  | CLIP-I | DINO | L1 | L2 |
|--------|--------|--------|--------|--------|
| CALCULATED | .928 | .901 | .057 | .014 |
| MENTIONED IN PAPER | .945 | .903 | .057 | .014 | 


After applying the transformation mentioned by you the values are as above. DINO is close but still not an exact match but CLIP is still not close. This is the exact implementation i followed:

Clip:
```
model, transform = clip.load('ViT-B/32', device)
image = transform(image).unqueeze(0).to(device)
```
Dino( referenced from magicbrush):
```
model = torch.hub.load('facebookresearch/dino:main', 'dino_vits16')
model.eval()
model.to(device)
transform = transforms.Compose([
   transforms.Resize(256, interpolation=3),
   transforms.CenterCrop(224),
   transforms.ToTensor(),
   transforms.Normalize((0.485, 0.456, 0.406),(0.229, 0.224, 0.225))])
image = transform(image).unsqueeze(0).to(device)
```


Can you specify other things that I need to take care.

**Comment by navvewas** [comment]
Sorry for the late reply, I'm not sure what caused this difference.  
A few options I can think of are:  
1. Different CLIP version in the load (we used `pip install git+https://github.com/openai/CLIP.git`).  
2. Different torchvision version used by the preprocessor (we used `torchvision==0.15.0`).  
3. Differences in the `eval_clip_i` function (ours is attached below).  
4. Different seed (since the rest of the results are the same, this is less likely).  

The eval CLIP function:
```python
def eval_clip_i(device, image_pairs, model, transform, metric='clip_i'):
    """
    Calculate the CLIP-I score: the cosine similarity between the generated image and the ground truth image.
    """
    def encode(image, model, transform):
        image_input = transform(image).unsqueeze(0).to(device)
        with torch.no_grad():
            if metric == 'clip_i':
                image_features = model.encode_image(image_input).detach().cpu().float()
            elif metric == 'dino':
                image_features = model(image_input).detach().cpu().float()
        return image_features

    eval_score = 0
    for img_pair in tqdm(image_pairs):
        generated_features = encode(Image.open(img_pair[0]).convert('RGB'), model, transform)
        gt_features = encode(Image.open(img_pair[1]).convert('RGB'), model, transform)
        similarity = 1 - spatial.distance.cosine(
            generated_features.view(generated_features.shape[1]),
            gt_features.view(gt_features.shape[1])
        )
        if similarity > 1 or similarity < -1:
            raise ValueError("Strange similarity value")
        eval_score = eval_score + similarity

    return eval_score / len(image_pairs)
```

Evaluating:
```python
model_CLIP, transform_CLIP = clip.load("ViT-B/32", device)
data = image_pairs
CLIP_score = eval_clip_i(device, image_pairs, model_CLIP, transform_CLIP)
```

Hopefully one of these is the cause of the differences.
